clippy_lints/src/non_copy_const.rs

```diff
@@ -197,12 +199,13 @@ impl<'tcx> NonCopyConst<'tcx> {
             // contained value.
             ty::Adt(def, ..) if def.is_union() => false,
             ty::Array(ty, _) => val
-                .unwrap_branch()
                 .iter()
                 .any(|field| Self::is_value_unfrozen_raw_inner(cx, *field, ty)),
             ty::Adt(def, args) if def.is_enum() => {
-                let (&variant_index, fields) = val.unwrap_branch().split_first().unwrap();
-                let variant_index = VariantIdx::from_u32(variant_index.unwrap_leaf().to_u32());
+                let (&ty::ValTree::Leaf(variant_index), fields) = val.split_first().unwrap() else {
```

Comment:
Wouldn't it be smarter to use `Some()` in the `let` since we already have an else case? 

```suggestion
                let Some((&ty::ValTree::Leaf(variant_index), fields)) = val.split_first() else {
```

Reply:
Fixed!